src/backend/tests/unit/services/tracing/test_tracing_service.py

```diff
@@ -0,0 +1,482 @@
+import asyncio
+import uuid
+from unittest.mock import MagicMock, patch
+
+import pytest
+from langflow.services.settings.base import Settings
+from langflow.services.settings.service import SettingsService
+from langflow.services.tracing.base import BaseTracer
+from langflow.services.tracing.service import (
+    TracingService,
+    component_context_var,
+    trace_context_var,
+)
+
+
+class MockTracer(BaseTracer):
+    def __init__(
+        self,
+        trace_name: str,
+        trace_type: str,
+        project_name: str,
+        trace_id: uuid.UUID,
+        user_id: str | None = None,
+        session_id: str | None = None,
+    ) -> None:
+        self.trace_name = trace_name
+        self.trace_type = trace_type
+        self.project_name = project_name
+        self.trace_id = trace_id
+        self.user_id = user_id
+        self.session_id = session_id
+        self._ready = True
+        self.end_called = False
+        self.get_langchain_callback_called = False
+        self.add_trace_list = []
+        self.end_trace_list = []
+
+    @property
+    def ready(self) -> bool:
+        return self._ready
+
+    def add_trace(
+        self,
+        trace_id: str,
+        trace_name: str,
+        trace_type: str,
+        inputs: dict[str, any],
+        metadata: dict[str, any] | None = None,
+        vertex=None,
+    ) -> None:
+        self.add_trace_list.append({
```

Comment:
I'm unable to mark the other lines through GitHub's interface.
But, this entire block needs to be indented as follows:
```suggestion
            self.add_trace_list.append(
            {
                "trace_id": trace_id,
                "trace_name": trace_name,
                "trace_type": trace_type,
                "inputs": inputs,
                "metadata": metadata,
                "vertex": vertex,
            }
        )
```